src/common/updating/updating.cpp

```
@@ -88,70 +88,78 @@ namespace updating
                     }
-                    // Do not break, since https://developer.github.com/v3/repos/releases/#list-releases
-                    // doesn't specify the order in which release object appear
-                    github_version = std::move(*extracted_version);
-                    release_object = std::move(potential_release_object);
                 }
-            }
-            else
-            {
-                const auto body = co_await client.request(Uri{ LATEST_RELEASE_ENDPOINT });
-                release_object = json::JsonValue::Parse(body).GetObjectW();
-                if (auto extracted_version = extract_version_from_release_object(release_object))
+                else
+                {
+                    const auto body = client.request(Uri{ LATEST_RELEASE_ENDPOINT }).get();
+                    release_object = json::JsonValue::Parse(body).GetObjectW();
+                    if (auto extracted_version = extract_version_from_release_object(release_object))
+                    {
+                        github_version = *extracted_version;
+                    }
+                }
+
+                if (github_version <= current_version)
                 {
-                    github_version = *extracted_version;
+                    return version_up_to_date{};
                 }
-            }
 
-            if (github_version <= current_version)
+                auto [installer_download_url, installer_filename] = extract_installer_asset_download_info(release_object);
+                return new_version_download_info{ extract_release_page_url(release_object),
+                                                     std::move(github_version),
+                                                     std::move(installer_download_url),
+                                                     std::move(installer_filename) };
+            }
+            catch (...)
             {
```

The catch block catches all exceptions (`catch (...)`) without any logging or error handling. In the original coroutine version, exceptions would propagate as part of the coroutine exception handling. Now they're silently swallowed and converted to NETWORK_ERROR. This could hide security-relevant errors (e.g., certificate validation failures, signature verification issues) and makes debugging harder. → Add logging for caught exceptions before converting to NETWORK_ERROR. Use Logger::error() to record the exception type and any available diagnostic information.